### ISSUE TITLE
No user session after login

### ISSUE THREAD

**Issue by mxfm** [systemd/systemd#16908]
**systemd version the issue has been seen with**
> 246.3-1

**Used distribution**
 > Arch linux

**Expected behaviour you didn't see**
 Just after login (I don't use any DE, plain startx started manually after login) there is no session created 
loginctl
> No sessions

loginctl list-users
> No users.

loginctl user-status
> Could not get properties: Unknown object '/org/freedesktop/login1/user/self'.

**Unexpected behaviour you saw**
Possibly relevant in journal during login:
"localhost login[1107]: pam_unix(login:session): session opened for user user(uid=1000) by LOGIN(uid=0)"
"localhost login[1107]: pam_systemd(login:session): Failed to get user record: Accessing a corrupted shared library"

**Steps to reproduce the problem**

Boot systemd 246.1-1 and pam 1.4.0-3, login at non-root account.

Important notes. There is session after root login, so there can be some permission issue. Groups of my user 'user' is in: tor vboxusers wireshark systemd-journal video lp input audio wheel user sudo. This issue was initially revealed when all polkit agents became broken with similar error message related to inability to connect to session bus. For example, error message of polkit-gnome is "(polkit-gnome-authentication-agent-1:1940): polkit-gnome-1-WARNING **: 06:52:55.689: Unable to determine the session we are in: No session for pid 1940" where 1940 is pid of polkit-gnome.

May be relevant.

*cat /etc/pam.d/system-login*

#%PAM-1.0

auth       required   pam_shells.so
auth       requisite  pam_nologin.so
auth       include    system-auth

account    required   pam_access.so
account    required   pam_nologin.so
account    include    system-auth

password   include    system-auth

session    optional   pam_loginuid.so
session    optional   pam_keyinit.so       force revoke
session    include    system-auth
session    optional   pam_motd.so          motd=/etc/motd
session    optional   pam_mail.so          dir=/var/spool/mail standard quiet
-session   optional   pam_systemd.so
session    required   pam_env.so           user_readenv=1

**Comment by mxfm** [comment]
Can be relevant. Perhaps initially bug appeared sometime around 10 August during pam/systemd update. The problem was that when I executed 'systemctl reboot' (or shutdown, or suspend) I got error message (not exact wording): "systemctl error: cannot connect to session bus". My initial guess was that some default permission rules in arch linux were changed, so I added following code to drop-in in '/etc/polkit/rules.d/':

polkit.addRule(function(action, subject) {
  if (action.id.match("org.freedesktop.login1.reboot") ||
      action.id.match("org.freedesktop.login1.hibernate") ||
      action.id.match("org.freedesktop.login1.suspend") || 
      action.id.match("org.freedesktop.login1.power-off")) {
    return polkit.Result.YES;
  }

However, only yesterday I noticed that session is broken after login, so it looks like I was fixing wrong issue.

**Comment by poettering** [comment]
Something is wrong with your installation, i.e. nss-systemd is older than the rest of your systemd installation.

Make sure that `/usr/lib/libnss_systemd.so.2` matches the rest of your systemd installation.

(ELIBAD is generated by systemd code when `libnss_systemd.so.2` is installed but lacks the `_nss_systemd_block` symbol intrdouced in systemd 246, see 037b0a47b0d7df09d720dda6703135117e7e0472)

**Comment by mxfm** [comment]
Oh, I was fooled by pacman (archlinux package manager) and my incorrect understanding of packaging systemd.

I was downgraded systemd versions and rolling back to v245 solved the issue. I was figuring out what was wrong - actually it was exactly as you written: libnss_systemd was out of sync with systemd. However, initially I doubted in this reason and spent 2-3 hours searching answer in different directions.

The issue was caused by follows (may be this is interesting to someone in the future). In June I was working at systemd patch and compiled git version of systemd called 'systemd-git'. In arch making this package produces two separate packages 'systemd-git' and 'systemd-libs-git'. Later the systemd package was switch back from 'systemd-git' to 'systemd' and then updated to v246. However, I forgot (did not noticed separate package) about systemd-libs and it stayed as 'systemd-libs-git' at v245 and it was not updated because it is a 'home-made' package. Despite the fact, that I should know this, the package manager could have been smarter.

tl;dr it is invalid issue.
